src/group_theory/divisible.lean

```diff
@@ -0,0 +1,266 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import group_theory.subgroup.pointwise
+import group_theory.group_action.pi
+import group_theory.quotient_group
+import algebra.group.pi
+
+/-!
+# Divisible Group and rootable group
+
+In this file, we define a divisible add monoid and a rootable monoid with some basic properties.
+
+## Main definition
+* An additive monoid `A` is said to be divisible by `α` iff for all `n ≠ 0 ∈ α` and `y ∈ A`, there
+  is an `x ∈ A` such that `n • x = y`. In this file, we adopt a constructive approach, i.e. we ask
+  for an explicit `div : A → α → A` function such that `div a 0 = 0` and `n • div a n = a` for all
+  `n ≠ 0 ∈ α`.
+* A monoid `A` is said to be rootable by `α` iff for all `n ≠ 0 ∈ α` and `y ∈ A`, there is an
+  `x ∈ A` such that `x^n = y`. In this file, we adopt a constructive approach, i.e. we ask for an
+  explicit `root : A → α → A` function such that `root a 0 = 1` and `(root a n)ⁿ = a` for all
+  `n ≠ 0 ∈ α`.
```

Comment:
You don't actually mention the name `add_monoid.divisible_by` here!

Should this just be called `divisible_by`? It seems actively confusing to use `add_monoid.divisible_by` to refer to divisibility in rings.

Reply:
I think so, `add_monoid.divisible_by` doesn't enable any dot notation or anything

Reply:
Don't forget to add to the docstring here,
```suggestion
## Main definition
* `divisible_by A α`: An additive monoid `A` is said to be divisible by `α` iff for all `n ≠ 0 ∈ α` and `y ∈ A`, there
  is an `x ∈ A` such that `n • x = y`. In this file, we adopt a constructive approach, i.e. we ask
  for an explicit `div : A → α → A` function such that `div a 0 = 0` and `n • div a n = a` for all
  `n ≠ 0 ∈ α`.
* `rootable_by A α`: A monoid `A` is said to be rootable by `α` iff for all `n ≠ 0 ∈ α` and `y ∈ A`, there is an
  `x ∈ A` such that `x^n = y`. In this file, we adopt a constructive approach, i.e. we ask for an
  explicit `root : A → α → A` function such that `root a 0 = 1` and `(root a n)ⁿ = a` for all
  `n ≠ 0 ∈ α`.
```
You'll need to wrap the lines

Reply:
I've gone ahead and fixed that. A lot of the other lemma names were inaccurate too.

Reply:
Thanks